### PR TITLE
Improve toolchain build

### PR DESCRIPTION
**Pull request by networkfusion**
* Change toolchain vars to ensure there are not issues with host OS
* Fix filename in build script for GMP
* Update docker actions to stop deprecation warnings.

### CODE REVIEW

File: .github/workflows/build-tool-windows.yml [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
 
-on: [push, pull_request]
+on:
+  # It is designed to only fire if the tools change.
+  push:
+    paths:
+      - 'tools/**'
+  pull_request:
+    paths:
+      - 'tools/**'
```
Comment (rasky) [inline .github/workflows/build-tool-windows.yml]: I’d rather not merge this. It makes an assumption about the absence of dependencies which is wrong in the unstable branch (the new mksprite depends on sprite.h, and rdpvalidate has dependencies too), and I don’t want to remember to update here any time we add one more. The workflow is quite fast anyway

Comment (networkfusion) [reply .github/workflows/build-tool-windows.yml]: Ok, I will revert that bit for the moment.

Comment (networkfusion) [reply .github/workflows/build-tool-windows.yml]: Fixed.

**Comment by rasky** [comment]
Merged upstream

**Pull request closed without merging**